pcap-rpcap.c

```diff
@@ -413,8 +419,31 @@ static int pcap_read_nocb_remote(pcap_t *p, struct pcap_pkthdr *pkt_header, u_ch
 		return -1;
 	}
 
+	/*
+	 * In the rpcap protocol, once the capture starts, the control socket isn't
+	 * used anymore until the capture ends.
+	 * However, it's the only way to check for connection errors
+	 * as the data socket may uses UDP.
+	 */
+	if (FD_ISSET(pr->rmt_sockctrl, &rfds)) {
+		uint8 byte;
+		const ssize_t nread = recv(pr->rmt_sockctrl, &byte, sizeof(byte), MSG_PEEK);
```

Comment:
This should call `sock_recv()`, to filter out UN\*X vs. Windows differences.  Note that `sock_recv()` returns an `int`; `ssize_t` is a UN\*X type but not a Windows type.

Reply:
Yep, I just saw the builder failure and exported it in the header before seeing your comment.

The problem with `sock_recv` (as quickly explained above) is that I should add another parameter just to handle `MSG_PEEK` or a special flag and that there are already special flags conflicting with this use case (like `SOCK_EOF*`). That seemed less clean to me at first :)

Reply:
>  I should add another parameter just to handle `MSG_PEEK` or a special flag

No, `SOCK_MSG_PEEK` already been added in d491c4d68d38902ce71fcceed49aeac1f574f533, so you don't have to add it, just use it.

Reply:
Thank you!